src/wp-includes/media.php

```diff
@@ -5614,3 +5846,83 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Determines whether to add `fetchpriority='high'` to loading attributes.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array $loading_attrs Array of the loading optimization attributes for the element.
+ * @param array $attr          Array of the attributes for the element.
+ * @return array Updated loading optimization attributes for the element.
+ */
+function wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr ) {
```

Comment:
We should also add `$tag_name` as a required parameter here, similar to how the public function requires it, as well as related to my comment for `wp_maybe_strip_loading_attribute()`. Let's add it as the 2nd parameter, before `$attr`.

We can then add a condition to this function to return early when `'img' !== $tag_name`.